games/rpg/src/lib.rs

```diff
@@ -0,0 +1,1483 @@
+                    weight: load,
+                    ability: dec!(0),
+                    ability_scaling: ability_scale,
+                    version: dec!(1),
+                };
+                let new_armor = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), armor_data));
+                
+                self.system_vault.authorize(|| 
+                    gold.burn());
+                self.system_vault.authorize(||
+                    resource1.burn());
+                self.system_vault.authorize(||
+                    resource2.burn());
+                self.system_vault.put(key_bucket);
+                new_armor
+            }
+            else {
+                let armor_data = Armor {  
+                    id: dec!(1),
+                    part: part,
+                    level: dec!(1), 
+                    health: hp,
+                    defense: def_bonus,
+                    weight: load,
+                    ability: dec!(0),
+                    ability_scaling: ability_scale,
+                    version: dec!(1),
+                };
+                let new_armor = self.system_vault.authorize(||
+                    borrow_resource_manager!(self.item_nft)
+                        .mint_non_fungible(&NonFungibleId::random(), armor_data));
+                
+                self.system_vault.authorize(|| 
+                    gold.burn());
+                self.system_vault.authorize(||
+                    resource1.burn());
+                self.system_vault.authorize(||
+                    resource2.burn());
+                self.system_vault.put(key_bucket);
+                new_armor
+            }
+        }
+        pub fn create_accessory_1(&mut self, gold: Bucket, resource1: Bucket, resource2: Bucket, part_chosen: u8,) -> Bucket {
+            assert!(gold.resource_address() == self.token_gold);
+            assert!(resource1.resource_address() == self.token_greavite);
+            assert!(resource2.resource_address() == self.token_wood);
+            let key_bucket: Bucket = self.system_vault.take(1);
+            let part = if part_chosen == 1 {
+                String::from("Gloves")
+            }
+            else if part_chosen == 2 {
+                String::from("Belt")
+            }
+            else {
+                String::from("Shoes")
+            };
+            let atk = dec!(".05") * self.seed_50();
+            let mag = dec!(".05") * self.seed_50();
+            let spd = dec!("5") * self.seed_50();
+            let atk_bonus = dec!(".1") * self.seed_50();
+            let mag_bonus = dec!(".1") * self.seed_50();
+            let spd_bonus = dec!("10") * self.seed_50();
+            let load = dec!(".01") * self.seed_50();
+            let ability_scale = dec!(0) * self.seed_50();
+            let rng = self.seed_3();
+            if rng == 1 {
```

Comment:
Same comment regarding the branching. Can it be made so that the branching either: a) happens at a different point so that there is not much code repetition, b) variables are used so that the branching happens for the variables but does not happen for the core code? 

As you can see, some part of the code are repeated a number of times throughout the if statement branches despite being the same, such as:
```rust
let new_accessory = self.system_vault.authorize(||
    borrow_resource_manager!(self.item_nft)
        .mint_non_fungible(&NonFungibleId::random(), accessory_data));

self.system_vault.authorize(|| 
    gold.burn());
self.system_vault.authorize(||
    resource1.burn());
self.system_vault.authorize(||
    resource2.burn());
self.system_vault.put(key_bucket);
new_accessory
```